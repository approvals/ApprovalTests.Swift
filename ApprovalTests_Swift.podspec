Pod::Spec.new do |s|
  s.name         = 'ApprovalTests_Swift'
  s.version      = '2.0.0'
  s.summary      = 'ApprovalTests for Swift, a powerful alternative to assertions'

  s.description = <<-DESC
                    Approval Tests are an alternative to assertions. Youâ€™ll find them useful for
                    testing objects with complex values (such as long strings), lots of properties,
                    or collections of objects.

                    ApprovalTests.Swift is compatible with the XCTest testing framework.
                  DESC

  s.homepage = 'https://github.com/approvals/ApprovalTests.Swift'
  s.license  = { :type => 'Apache License, Version 2.0', :file => 'LICENSE' }
  s.author   = { 'Jon Reid' => 'jon@qualitycoding.org' }
  s.social_media_url = 'https://twitter.com/qcoding'

  s.swift_version = '5.0'
  s.ios.deployment_target = '12.0'
  s.osx.deployment_target = '10.15'

  s.source = { :git => 'https://github.com/approvals/ApprovalTests.Swift.git', :tag => 'v2.0.0' }
  s.framework = 'XCTest'
  s.source_files      = "ApprovalTests.Swift/iOSApprovalsWatcher.py", "ApprovalTests.Swift/**/*.swift" 
  s.ios.exclude_files = "ApprovalTests.Swift/Reporters/macOS/*.swift"
  s.osx.exclude_files = "ApprovalTests.Swift/Reporters/iOS/*.swift"

end
