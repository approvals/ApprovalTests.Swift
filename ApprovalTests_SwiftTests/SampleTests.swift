#if os(iOS)
  import ApprovalTests_iOS
#else
  import ApprovalTests_Swift
#endif
import XCTest

final class SampleTests: XCTestCase {
    private var getWords = ["Approval", "Tests"]

    func testString() throws {
        var s = "Approval"
        s += "Tests"
        try Approvals.verify(s)
    }

    func testObject() throws {
        let myObject = MyClass()
        try Approvals.verify(String(describing: myObject))
    }

    func testArray() throws {
        try Approvals.verifyAll(getWords, label: "Text")
    }

    func testArray2() throws {
        try Approvals.verifyAll(header: "words", getWords)
    }

    func testArrayWithLabeler() throws {
        try Approvals.verifyAll(header: "uppercased", getWords) { w in
            "\(w) -> \(w.uppercased())"
        }
    }

    func testArrayWithLabeler2() throws {
        try Approvals.verifyAll(header: "", getWords, label: "text")
    }
}

private class MyClass: CustomStringConvertible {
    var value = "ApprovalTests"

    var description: String { "MyClass: \(value)" }
}
