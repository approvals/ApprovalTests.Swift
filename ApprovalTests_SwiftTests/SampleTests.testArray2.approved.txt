words

[0] = Approval
[1] = Tests
